upgrades/steps_28.go

```diff
@@ -200,6 +202,14 @@ func iaasUniterState(ctx Context, unitNames []string) ([]string, error) {
 			fileNames.Add(storageData.filename)
 		}
 
+		relationData, err := readRelationState(uniterStateDir)
+		if err != nil && !os.IsNotExist(err) && !errors.IsNotFound(err) {
```

Comment:
Since we control `readRelationState` why not have it test for `os.IsNotExist` internally and return back errors.IsNotFound? An example of this pattern can be found [here](https://github.com/juju/juju/blob/develop/worker/meterstatus/state.go#L112).

If you do this, can you also please change `readStorageState` above to use the same pattern?

Reply:
for the relation piece  IsNotFound will not occur.  it's specifically returned in the readStorageState though.  I can replace the NotExists errors with NotFound for storage.